backend/fms_core/services/sample.py

```diff
@@ -1048,4 +1049,24 @@ def can_remove_sample(sample: Sample) -> Tuple[bool, List[str], List[str]]:
         was_processed = ProcessMeasurement.objects.filter(source_sample_id=sample.id).exists()
         is_removable = not is_child and not is_parent and not was_processed
 
-    return is_removable, errors, warnings
\ No newline at end of file
+    return is_removable, errors, warnings
+
+T = TypeVar("T")
+def defaultSelection_exceptedIDs(queryset: QuerySet[T], default_selection: bool, excepted_ids: list[int]) -> QuerySet[T]:
+    """
+    Filter a queryset based on the default_selection and excepted_ids.
+    If default_selection is True, the queryset will exclude the excepted_ids.
```

Comment:
This goes against what i would expect by the name. default selection would indicate a typical filter rather than an exclusion you seem to be applying your front end logic to the backend name scheme and it makes it harder to understand. This little function is barely useful as is if you have to read it to understand it you are removing all gain.
I strongly suggest you just add those lines to your viewset they need context and this function bring the worse part of frontend design to the backend (scattering the code all over the place).